pkg/pubsub/provider/provider.go

```diff
@@ -0,0 +1,47 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/connection"
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/dapr"
+)
+
+var pubSubs = newPubSubSet(map[string]InitiateConnection{
+	dapr.Name: dapr.NewConnection,
+},
+)
+
+type pubSubSet struct {
+	supportedPubSub map[string]InitiateConnection
+}
+
+// returns new client for pub sub tool.
+type InitiateConnection func(ctx context.Context, config interface{}) (connection.Connection, error)
+
+func newPubSubSet(pubSubs map[string]InitiateConnection) *pubSubSet {
+	supported := make(map[string]InitiateConnection)
+	set := &pubSubSet{
+		supportedPubSub: supported,
+	}
+	for name := range pubSubs {
+		set.AddSupportedTool(name, pubSubs[name])
+	}
+	return set
+}
+
+func (ps *pubSubSet) AddSupportedTool(name string, new InitiateConnection) {
+	if _, ok := ps.supportedPubSub[name]; ok {
+		panic(fmt.Sprintf("pubsub %v registered twice", name))
```

Comment:
how do we get here?

Reply:
removed this code, it was unreachable and I missed it in the changes